proj/hps_accel/src/tensorflow/lite/kernels/internal/reference/integer_ops/conv_accel.cc

```diff
@@ -94,6 +107,14 @@ void ConvPerChannel4x4(
         hps_accel::LoadInput(input_width, input_depth, current_input_data);
 
         for (int out_channel = 0; out_channel < output_depth; ++out_channel) {
+          if (filter_load_needed &&
```

Comment:
I think this commit should go in *as-is*, because it unblocks other work, but wanted to make a note about an alternate implementation:

Given `N = ceil(output_depth / out_channels_per_filter_load)` and `M = out_x * out_y * input_depth` then:

- this commit requires `M` input loads and `NM` filter loads.
- alternative: if we process all the input once per filter load, we instead get `NM` input loads, but only `N` filter loads. 

This alternative is still the same processing complexity (i.e `O(NM)`), but given that input values come from RAM and filter values come from SPI there is potential for significant difference. We should test performance against the models we plan to run.

mnv2_first implements this alternative, and it is quite complex to efficiently calculate the index into output_data.

Reply:
Yes I wondered about rearranging things here so that we could further reduce the number of filter loads for the large layers. But as you say, it makes the indexes trickier to compute. I went with this version first as it required essentially no changes to the structure of the loops or the index lookups.